pkg/asset/util/util.go

```diff
@@ -0,0 +1,67 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/kubeconfig"
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/tls"
+)
+
+// GetTypeFromString takes the asset type as string and returns an actual instance of that type
+func GetTypeFromString(assetType string) (asset.WritableAsset, error) {
```

Comment:
Why is this required?

Reply:
Because we dump a map where the type is the key. And when we unmarshal we cannot dynamically construct the object in golang. The primary type of Asset is interface{}, if it were a struct we could have made a custom json unmarshal function. This is the most compact way I can think of in the current scenario.

Reply:
https://golang.org/pkg/reflect/#TypeOf ? why can't we use this.

Reply:
Can't use TypeOf, because all we have is the string of the Type. reflect.Type cannot be put into json unless we transform it to a string. How do we get the actual type from the 'string'? We have to hardcode a typemap.
For this exact reason, this PR last week had the asset.Name() as the key so that we do not need to implement this hardcoded mapper. It was brought down because Name() will be forced to be unique && something that cannot be generated at runtime. (e.g. certkey).

Reply:
https://golang.org/pkg/reflect/#Type
```go
        // Name returns the type's name within its package for a defined type.
        // For other (non-defined) types it returns the empty string.
        Name() string
```

Reply:
When I tested this, all our object types return the empty string. Some valid, but undiscovered theory there.

Reply:
All the assets are pointers. That's why they have empty return from Name....

Check doc for Elem to get underlying type for pointer.